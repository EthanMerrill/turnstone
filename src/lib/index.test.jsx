import React from 'react'
import renderer from 'react-test-renderer'

import { vi, describe, expect, test } from 'vitest'
import Turnstone from './index'

vi.mock('./components/container', () => ({ default: () => 'Container' }))

const listbox = {
  data: ['foo', 'bar', 'foobar'],
  searchType: 'startswith'
}

const component = renderer.create(
  <Turnstone listbox={listbox} />
)

const tree = component.toJSON()

describe('Turnstone', () => {
  test('Turnstone component renders correctly', () => {
    expect(tree).toMatchSnapshot()
  })

  test('Turnstone component passes all props to Container component along with default props', () => {
    const props = {...component.root.children[0].children[0].props}

    // The id prop is randomly generated so must be excluded
    delete props.id

    expect(props).toEqual({
      autoFocus: false,
      clearButton: false,
      clearButtonAriaLabel: 'Clear contents',
      clearButtonText: 'Ã—',
      debounceWait: 250,
      defaultListboxIsImmutable: true,
      disabled: false,
      listbox,
      listboxIsImmutable: true,
      maxItems: 10,
      minQueryLength: 1,
      placeholder: ''
    })
  })
})