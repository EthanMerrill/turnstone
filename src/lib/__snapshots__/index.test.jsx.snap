// Vitest Snapshot v1

exports[`Turnstone > Turnstone component renders correctly 1`] = `"Container"`;
